gctree/isotyping.py

```diff
@@ -159,6 +160,17 @@ def resolutions(self) -> Sequence["Isotype"]:
         else:
             return [self.copy()]
 
+    def mutate(self, lambda0=0.01):
+        """Returns a new isotype object, which may be of the same class, or may be
+        a later class.
+        """
+        newisotype = self.copy()
+        if scipy.random.poisson(lambda0) > 0:
+            n = len(self.order) - newisotype.isotype
+            if n != 1:
+                newisotype.isotype = self.isotype + scipy.random.randint(1, n)
+        return newisotype
```

Comment:
I'm not sure about the Poisson rv here—you are using it to make a binary choice? Shouldn't `n` be based on the Poisson draw, rather and drawn uniformly (but then truncated based on the terminal isotype)?

I would also suggest documenting `lambda0` in the docstring.

It's also implicit here that this function simulates for one unit of simulation time, so I wonder if it would be better to put this code in the `mutation_model.MutationModel.simulate` function?